docs/tutorial-run-solver.md

```diff
@@ -0,0 +1,176 @@
+
+```sh
+git clone git@github.com:gnosis/dex-solver.git
+```
+
+## Building The Driver
+
+Build the driver (with open-solver dockerhub image) from within the root directory of `dex-services` repo. This command builds the driver while simultanesouly baking the open solver into the the build.
+
+```sh
+docker-compose build --build-arg SOLVER_BASE=gnosispm/dex-open-solver:master stablex-debug
+```
+
+Note that, if `SOLVER_BASE` is not specified, the driver's own internal `naive-solver` will be used. Alternatively, if you have your own personalized solver, this image would need to be specified during the build.
+We will see that specifying solver type again at runtime is also expected and this may seem counterintuitive.
+However, the argument used at runtime is meant to specify potentially different implementations from within the existing solver build.
+For example, you may have a `standard-solver` which matches orders according to one algorithm and another, say `fallback-solver` which makes use of a fixed list of currated tokens or simply operates in a different way.
+
+## Configuring Environment Variables
+
+The following three environment variables are required when running the dex-services project.
+Note that the account corresponding to the private key should be funded with sufficient ETH to pay gas for solution submission.
+
+```sh
+export PRIVATE_KEY=<YOUR_PRIVATE_KEY>
+export ETHEREUM_NODE_URL=https://mainnet.infura.io/v3/<YOUR_INFURA_KEY>
+export NETWORK_ID=<YOUR_PREFERED_NETWORK_ID>
+```
+
+By placing these values into an `.env` file and sourcing this file, you can avoid adding them to as runtime arguments on every execution.
+
+## Acquiring the Orderbook
+
+It can take a very long time for the driver to load the orderbook and this process is also limited by infura rate limits so it is best to supply an existing orderbook file, available for download at
+
+- [mainnet](https://gnosis-dfusion-volume-mainnet.s3.amazonaws.com/stablex_orderbook_mainnet.bin)
+- [rinkeby](https://gnosis-dfusion-volume-rinkeby.s3.amazonaws.com/stablex_orderbook_rinkeby.bin)
+
+Copy this orderbook file into `dex-services/target` and point your solver to this file via the runtime argument `--orderbook-file` or environment vairable `ORDERBOOK_FILE`. That is, you can append `ORDERBOOK_FILE=/app/dex-services/target/stablex_orderbook_mainnet.bin` to your `.env` file.
+
+## Run the Solver
+
+First run and enter the solver container
+
+```sh
+docker-compose run -v $(PWD)/:/app/dex-services stablex-debug
+```
+
+Since the project was mounted inside the container you can make changes to the driver code and they will be directly reflected on every restart.
+
+### Without orderbook file (not recommended)
+
+```sh
+cargo run --bin driver -- --solver-type open-solver --network-id $NETWORK_ID --node-url $ETHEREUM_NODE_URL --private-key $PRIVATE_KEY
+```
```

Comment:
```
error: 'open-solver' isn't a valid value for '--solver-type <solver-type>'
	[possible values: BestRingSolver, NaiveSolver, OpenSolver, StandardSolver]

	Did you mean 'OpenSolver'?
```
Since it's not recommended, I'd put the recommended way to do it first. It could also be a subsection of the next section, after changing the titles.

Reply:
Very strange. I will look into this.

Reply:
Yes, this was a freshly introduced config change introduced just last week. 

https://github.com/gnosis/dex-services/pull/1413

I will make the necessary changes.